01-Authorization/auth0authorization/utils.py

```diff
@@ -1,9 +1,10 @@
-import json
-import os
+from urllib.error import URLError
 
-from django.contrib.auth import authenticate
 import jwt
-import requests
+
+from django.contrib.auth import authenticate
+from django.conf import settings
+from rest_framework.exceptions import AuthenticationFailed
 
 
 def jwt_get_username_from_payload_handler(payload):
```

Comment:
Calling the `authenticate` backend to insert the user in the DB in this method as a side-effect of getting the username from the token payload is confusing.